yt_dlp/utils/_utils.py

```diff
@@ -4763,6 +4763,10 @@ def jwt_decode_hs256(jwt):
     return json.loads(base64.urlsafe_b64decode(f'{payload_b64}==='))
 
 
+def jwt_is_expired(token, buffer=300, key='exp'):
+    return jwt_decode_hs256(token)[key] - time.time() < buffer
```

Comment:
a non-fatal version would look like this
```suggestion
    exp = traversal.traverse_obj(token, ({jwt_decode_hs256}, key, {int, float})) or 0
    return exp - time.time() < buffer
```